### PR TITLE
Slot handling redesign

### PR DESCRIPTION
**Pull request by sebhz**
Slot handling (mostly) moved in its own file to make cap32.cpp a bit lighter.
Only xxx_file now used (except for cartridge, which is sort of a hack).

### CODE REVIEW

File: src/slotshandler.cpp [hunk 1]
```diff
@@ -103,6 +103,12 @@ int file_size (int file_num)
    }
 }
 
+#define FILL_SLOT_INFO(filevar, processedvar, file) \
```
Comment (ColinPitrat) [inline src/slotshandler.cpp]: You can do the same (but with the benefit of a function and without the issues of macros) with:
void fill_slot_info(std::string &filevar, bool &processedvar, const string& fullpath)
{
  filevar = fullpath;
  processedvar = true;
}

You could also add the extension and factorize the if & logging part.

Comment (sebhz) [inline src/slotshandler.cpp]: Thx. Take your time for the review :-)

File: src/cap32.h [hunk 2]
```diff
@@ -266,22 +259,22 @@ typedef struct {
    unsigned int kbd_layout;
 
    unsigned int max_tracksize;
-   std::string snap_path;
-   std::string snap_file;
-   bool snap_zip;
+
+   std::string snap_path; // Path where images will be loaded/saved by default. Only one exposed in CFG file
```
Comment (ColinPitrat) [inline src/cap32.h]: Replace 'images' by 'snapshot' in comments ? (not necessarly less confusing ... how to ensure there is no confusion with screenshot ?)

File: src/log.cpp [hunk 3]
```diff
@@ -1,3 +1,3 @@
 #include "log.h"
 
-bool log_verbose = false;
+bool log_verbose = true;
```
Comment (ColinPitrat) [inline src/log.cpp]: You probably don't want to commit that (this is very verbose).
Note for another patch: now that we have flags parsing, we should put a `--verbose` flag for it.

File: src/slotshandler.cpp [hunk 4]
```diff
@@ -1347,20 +1352,10 @@ int tape_insert_voc (FILE *pfile)
...
-          fclose(file);
-        }
-      } else {
-        if(cpr_load(CPC.cart_path + CPC.cart_file) != 0) {
-          fprintf(stderr, "Load of cartridge failed. Aborting.\n");
-          exit(-1);
-        }
-      }
-    }
+     if (file_load(CPC.cart_path + CPC.cart_file, OTHER)) {
```
Comment (ColinPitrat) [inline src/slotshandler.cpp]: why not just always set CPC.cart_file to the full path ? And use path only for Load/Save menu as for others.

File: src/slotshandler.cpp [hunk 5]
```diff
@@ -116,65 +122,64 @@ void fillSlots (std::vector<std::string> slot_list, t_CPC& CPC)
...
+            FILL_SLOT_INFO(CPC.cart_file, have_CPR, fullpath);
+            continue;
          }
       }
    }
 }
 
+void loadSlots(void) {
+   memset(&driveA, 0, sizeof(t_drive)); // clear disk drive A data structure
+   file_load(CPC.drvA_file, DSK_A);
```
Comment (ColinPitrat) [inline src/slotshandler.cpp]: Should we check error codes here ?
Or maybe rather LOG_ERROR in file_load ...

**Comment by sebhz** [comment]
OK - some more commits to address your review. ROM handling is a bit fishy still, because we are actually using ROM path for this. Apart from this we should be clean.
The zip opening/handling can probably be factorized away also - I'll have a look into this one day.

File: src/slotshandler.cpp [hunk 6]
```diff
@@ -103,9 +103,16 @@ int file_size (int file_num)
    }
 }
 
-#define FILL_SLOT_INFO(filevar, processedvar, file) \
-    (filevar) = (file); \
-    (processedvar) = true;
+inline bool fillSlot(std::string &filevar, bool &processedvar, const std::string fullpath, std::string extension, const std::string type_ext, const std::string type_desc) {
```
Comment (ColinPitrat) [inline src/slotshandler.cpp]: To avoid a copy, you should prefer passing the strings by reference (e.g const std::string& extension)

**Comment by ColinPitrat** [comment]
Thanks for this, this is a lot less duplication and more user friendly too !

**Pull request merged**